bin/microstrax

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+import argparse
+
+import hug
+import strax
+import straxen
+
+st: strax.Context
+
+
+def load_context(name):
+    global st
+    st = getattr(straxen.contexts, name)()
+    st.context_config['forbid_creation_of'] = (
+            '*',
+            # In case someone has an old strax that doesn't support * here:
+            'raw_records', 'records', 'peaklets',
+            'peaks', 'event_basics', 'event_info')
+
+
+@hug.exception(Exception)
+def handle_exception(exception):
+    return {'error': str(exception)}
+
+
+@hug.get('/get_data')
+def get_data(
+        run_id: hug.types.text,
+        target: hug.types.text,
+        max_n: hug.types.number = 1000,
+        start: hug.types.float_number = None,
+        end: hug.types.float_number = None,
+        selection_str: hug.types.text = None):
+    try:
+        st
+    except NameError:
+        return "Context not loaded???"
+
+    if max_n <= 0:
+        max_n = None
+    t0 = st.estimate_run_start(run_id, target)
+    if start is None or end is None:
+        time_range = None
+    else:
+        time_range = [t0 + int(start * int(1e9)),
+                      t0 + int(end * int(1e9))]
+
+    if not max_n:
+        load_bytes = st.size_mb(run_id, target) * int(1e6)
+
+    else:
+        # Let's see how much we really need to load, given max_n.
+        md = st.get_metadata(run_id, target)
+        if time_range is None:
+            time_range = [md['start'], md['end']]
+        load_at_least_n = 0
+        load_bytes = 0
+        for chunk_info in md['chunks']:
+
+            if (chunk_info['start'] > time_range[1]
+                    or chunk_info['end'] < time_range[0]):
+                # None of this chunk will be loaded
+                continue
+            load_bytes += chunk_info['nbytes']
+
+            if (chunk_info['start'] > time_range[0]
+                    and chunk_info['end'] < time_range[1]):
+                # All of this chunk will be loaded
+                load_at_least_n += chunk_info['n']
+
+            if load_at_least_n > max_n:
+                # No point loading any more, we already exceeded max_n
+                time_range[1] = chunk_info['end']
+                break
+
+    if load_bytes > int(1e9):
+        raise ValueError(f"Cannot load {target} for run {run_id}: "
```

Comment:
Is raising an error the most robust way to signal to the caller that this doesn't work, or should we rather return something like `{'message' : "Cannot load..."}`, so the the caller knows why it didn't get anything when it asked? I've spent more hours than was probably necessary tracing 500 errors because the server code had a bug, and unanswered requests seems like a great way to generate these.

Reply:
There is an exception handler defined, so if there is an error you will get {'error': 'something failed'} already. We could consider also returning the exception type and traceback though.

Reply:
Ok, that should be fine